app/src/test/java/io/homeassistant/companion/android/onboarding/integration/MobileAppIntegrationPresenterImplSpec.kt

```diff
@@ -0,0 +1,65 @@
+package io.homeassistant.companion.android.onboarding.integration
+
+import io.homeassistant.companion.android.domain.integration.IntegrationUseCase
+import io.mockk.*
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.test.resetMain
+import kotlinx.coroutines.test.setMain
+import org.spekframework.spek2.Spek
+import org.spekframework.spek2.style.specification.describe
+import java.lang.Exception
+
+
+object MobileAppIntegrationPresenterImplSpec : Spek({
+
+    beforeEachTest {
+        Dispatchers.setMain(Dispatchers.Unconfined)
+    }
+
+    afterEachTest {
+        Dispatchers.resetMain()
+    }
+
+    describe("presenter") {
+        val integrationUseCase by memoized { mockk<IntegrationUseCase>(relaxUnitFun = true) }
+        val view by memoized { mockk<MobileAppIntegrationView>(relaxUnitFun = true) }
+        val presenter by memoized { MobileAppIntegrationPresenterImpl(view, integrationUseCase) }
+
+        describe("on click skip") {
+            beforeEachTest {
+                presenter.onSkip()
+            }
+
+            it("should continue") {
+                coVerifyAll { view.registrationSkipped() }
+            }
+        }
+
+        describe("on click retry") {
+
+            it("should try to register and succeed") {
+                coEvery {integrationUseCase.registerDevice(any())} just runs
```

Comment:
here you should avoid any and test the real value (you can mock static the BuildConfig) see https://mockk.io/

Reply:
I wasn't able to mock it out, I looked into it and found other people wrapped the calls in functions to get around it.  Not sure if we want to do that.  I was able to get rid of the any though.

Reply:
[Here](https://github.com/home-assistant/home-assistant-android/blob/3219504a122be600ba8c76af2888ece962ee78c9/app/src/test/java/io/homeassistant/companion/android/onboarding/authentication/AuthenticationPresenterImplSpec.kt#L47) I was able to mock the Uri from Android.
Did you try something like this ?
````
mockkStatic(BuildConfig::class)
every { BuildConfig. VERSION_NAME } returns "1.0.0"

Reply:
Tried exactly that and it failed.